src/aiovodafone/api.py

```diff
@@ -38,6 +38,28 @@ class VodafoneStationDevice:
 class VodafoneStationCommonApi(ABC):
     """Common API calls for Vodafone Station routers."""
 
+    @staticmethod
+    async def get_device_type(host: str) -> DeviceType:
+        """Finds out the device type of a Vodafone Stations and returns it as enum.
+        The Technicolor devices always answer with a valid HTTP response, the
+        Sercomm returns 404 on a missing page. This helps to determine which we are
+        talking with.
+
+        Args:
+            host (str): The router's address, e.g. `192.168.1.1`
+
+        Returns:
+            DeviceType: If the device is a Technicolor, it returns
+            `DeviceType.TECHNICOLOR` otherwise `DeviceType.SERCOMM`.
+        """
+        async with aiohttp.ClientSession() as session:
+            async with session.get(f"http://{host}/api/v1/login_conf") as response:
+                if response.status == 200:
+                    response_json = await response.json()
+                    if "data" in response_json and "ModelName" in response_json["data"]:
+                        return DeviceType.TECHNICOLOR
+                return DeviceType.SERCOMM
```

Comment:
We should check if login.html exist before saying it's SERCOMM. A user could always put the wrong ip.

Reply:
I added an extra check for `login.html`. This introduces a new return type: `None` if neither types fit.